token_dispenser_client/token_dispenser_client.py

```diff
@@ -0,0 +1,216 @@
+
+
+def get_parameter_by_name(name: str, with_decryption:bool = False)->str:
+    try:
+        if not name.endswith('/'):  # Check if it's a specific parameter name
+            response = ssm.get_parameter(
+                Name=name,
+                WithDecryption=with_decryption
+            )
+        return response['Parameter']['Value']
+    except KeyError:
+        print(f"Missing 'Parameter' or 'Value' key with name: {name} response:{response} "
+                       f"with_decryption:{with_decryption}")
+        raise KeyError(f"Missing 'Parameter' or 'Value' key with name: {name} response:{response} "
+                       f"with_decryption:{with_decryption}")
+    except Exception as e:  # Catch any other error.
+        print(f"An unexpected error occurred during get_parameter: name: {name} response:{response} "
+                       f"with_decryption:{with_decryption}")
+        raise e
+
+
+def get_parameters_by_path(path:str, with_decryption=True):
+    """
+    Retrieves parameters from AWS Systems Manager Parameter Store by path.
+
+    Args:
+        path (str): The path to the parameters.
+        with_decryption (bool, optional): Whether to decrypt SecureString parameters. Defaults to True.
+
+    Returns:
+        dict: A dictionary containing the parameter names and values, or None if an error occurs.
+        Returns empty dict if no parameters found on the path.
+    Raises:
+        botocore.exceptions.ClientError: If an error occurs while calling the SSM API.
+    """
+    try:
+        parameters = {}
+        next_token = None
+        while True:
+            if next_token:
+                response = ssm.get_parameters_by_path(
+                    Path=path,
+                    Recursive=True,
+                    WithDecryption=with_decryption,
+                    NextToken=next_token
+                )
+            else:
+                response = ssm.get_parameters_by_path(
+                    Path=path,
+                    Recursive=True,
+                    WithDecryption=with_decryption
+                )
+
+            for param in response.get('Parameters', []):
+                parameters[param['Name']] = param['Value']
+
+            next_token = response.get('NextToken')
+            if not next_token:
+                break
+
+        return parameters
+    except(ssm.exceptions.InternalServerError, ssm.exceptions.InvalidFilterKey, ssm.exceptions.InvalidFilterOption,
+           ssm.exceptions.InvalidFilterValue, ssm.exceptions.InvalidKeyId, ssm.exceptions.InvalidNextToken) \
+            as ssm_exception:
+        print(f'ssm get_parameters_by_path error with path: {path} with_decryption:{with_decryption} '
+              f'error: {ssm_exception}')
+        raise ssm_exception
+    except Exception as e:
+        print(f'Unexpected error in get_parameters_by_path with path: {path} with_decryption:{with_decryption} '
+              f'error: {e}')
+        raise e
```

Comment:
We don't need to catch the exception just to log it and reraise it to the next level. We can just let it boil up the chain or catch it and replace it with our own error class